module-system/module-implementations/sov-chain-state/src/lib.rs

```diff
@@ -0,0 +1,144 @@
+#![deny(missing_docs)]
+#![doc = include_str!("../README.md")]
+
+/// Contains the call methods used by the module
+pub mod call;
+
+/// Genesis state configuration
+pub mod genesis;
+
+/// Hook implementation for the module
+pub mod hooks;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(test)]
+mod tests_helpers;
+
+#[cfg(feature = "native")]
+/// The query interface with the module
+pub mod query;
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use sov_modules_api::Error;
+use sov_modules_macros::ModuleInfo;
+use sov_rollup_interface::mocks::TestValidityCond;
+use sov_rollup_interface::zk::{ValidityCondition, ValidityConditionChecker};
+use sov_state::WorkingSet;
+
+#[derive(BorshDeserialize, BorshSerialize, Clone, Debug, PartialEq, Eq)]
+/// Structure that contains the information needed to represent a single state transition.
+pub struct StateTransitionId<Cond: ValidityCondition> {
+    da_block_hash: [u8; 32],
+    post_state_root: [u8; 32],
+    validity_condition: Cond,
+}
+
+impl StateTransitionId<TestValidityCond> {
```

Comment:
nit: Let's add a comment explaining why `new` is only available for `TestValidityCond`